.github/workflows/ci.yml

```diff
@@ -0,0 +1,97 @@
+name: CI
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  test-on-ubuntu:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04]
+        compiler: [{C: gcc-7, CXX: g++-7}, {C: gcc-11, CXX: g++-11}]
+        python: [python3.8, python3.9, python3.10]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # - Installing software-properties-common adds add-apt-repository
+      - name: Update install tools
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y software-properties-common curl
+
+      # - ppa:ubuntu-toolchain-r/test is needed to have compiler available on all ubuntu versions
+      # - Installing g++-x also installs gcc-x
+      - name: Install C and CXX compilers
+        run: |
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+          sudo apt-get install -y ${{ matrix.compiler.CXX }}
+
+      # - ppa:deadsnakes/ppa is needed to have python3.x-dev available on all ubuntu versions
+      # - python3.x-dev is used instead of python3.x because the -dev version includes python header
+      # files needed to compile C extension modules
+      - name: Install python
+        run: |
+          sudo add-apt-repository -y ppa:deadsnakes/ppa
+          sudo apt-get install -y ${{ matrix.python }}-dev ${{ matrix.python }}-distutils
+          curl https://bootstrap.pypa.io/get-pip.py | ${{ matrix.python }}
+          pip install --upgrade setuptools
+
+      - name: Install general build dependencies
+        run: |
+            sudo apt-get install -y \
+              doxygen \
+              libgmp-dev \
+              pandoc
+
+      - name: Install build dependencies for SymForce benchmarks
+        run: |
+            sudo apt-get install -y \
+              libboost-dev \
+              libboost-serialization-dev \
+              libboost-system-dev \
+              libboost-filesystem-dev \
+              libboost-thread-dev \
+              libboost-program-options-dev \
+              libboost-date-time-dev \
+              libboost-timer-dev \
+              libboost-chrono-dev \
+              libboost-regex-dev \
+              libgoogle-glog-dev \
+              libeigen3-dev
+
+      - name: Install python dependencies
+        run: pip install -r dev_requirements.txt
+
+      - name: Run cmake build
+        run: |
+          mkdir build
+          cd build
+          cmake .. \
+            -D CMAKE_C_COMPILER=${{ matrix.compiler.C }} \
+            -D CMAKE_CXX_COMPILER=${{ matrix.compiler.CXX }} \
+            -D SYMFORCE_PYTHON_OVERRIDE=${{ matrix.python }} \
+            -D SYMFORCE_BUILD_BENCHMARKS=ON
+          make -j $(nproc)
```

Comment:
I'd prefer we do the canonical `cmake -B build -D ...` and `cmake --build build -j $(nproc)` here instead of the `mkdir build; cd build; cmake .. ...; make ...`